epoxy-processor/src/main/java/com/airbnb/epoxy/EpoxyProcessor.java

```diff
@@ -365,7 +381,7 @@ private MethodSpec buildEquals(ClassToGenerateInfo helperClass) {
         .beginControlFlow("if (o == this)")
         .addStatement("return true")
         .endControlFlow()
-        .beginControlFlow("if (!(o instanceof $T))", helperClass.getOriginalClassNameWithoutType())
+        .beginControlFlow("if (!(o instanceof $T))", helperClass.getGeneratedName())
```

Comment:
Use the generated name in the instance of, so when we cast below this we won't get a class cast exception